lib/openstudio-standards/standards/Standards.CoilCoolingWaterToAirHeatPumpEquationFit.rb

```diff
@@ -0,0 +1,109 @@
+class Standard
+  # @!group CoilCoolingWaterToAirHeatPumpEquationFit
+
+  # Finds capacity in W
+  #
+  # @return [Double] capacity in W to be used for find object
+  def coil_cooling_water_to_air_heat_pump_find_capacity(coil_cooling_water_to_air_heat_pump)
+    capacity_w = nil
+    if coil_cooling_water_to_air_heat_pump.ratedTotalCoolingCapacity.is_initialized
+      capacity_w = coil_cooling_water_to_air_heat_pump.ratedTotalCoolingCapacity.get
+    elsif coil_cooling_water_to_air_heat_pump.autosizedRatedTotalCoolingCapacity.is_initialized
+      capacity_w = coil_cooling_water_to_air_heat_pump.autosizedRatedTotalCoolingCapacity.get
+    else
+      OpenStudio.logFree(OpenStudio::Warn, 'openstudio.standards.CoilCoolingWaterToAirHeatPumpEquationFit', "For #{coil_cooling_water_to_air_heat_pump.name} capacity is not available, cannot apply efficiency standard.")
+      return 0.0
+    end
+
+    return capacity_w
+  end
+
+  # Finds lookup object in standards and return efficiency
+  #
+  # @param rename [Bool] if true, object will be renamed to include capacity and efficiency level
+  # @return [Double] full load efficiency (COP)
+  def coil_cooling_water_to_air_heat_pump_standard_minimum_cop(coil_cooling_water_to_air_heat_pump, rename = false)
+    search_criteria = {}
+    search_criteria['template'] = template
+    capacity_w = coil_cooling_water_to_air_heat_pump_find_capacity(coil_cooling_water_to_air_heat_pump)
+    capacity_btu_per_hr = OpenStudio.convert(capacity_w, 'W', 'Btu/hr').get
+    capacity_kbtu_per_hr = OpenStudio.convert(capacity_w, 'W', 'kBtu/hr').get
+
+    # Look up the efficiency characteristics
+    coil_props = model_find_object(standards_data['water_source_heat_pumps'], search_criteria, capacity_btu_per_hr, Date.today)
+
+    # Check to make sure properties were found
+    if coil_props.nil?
+      OpenStudio.logFree(OpenStudio::Warn, 'openstudio.standards.CoilCoolingWaterToAirHeatPumpEquationFit', "For #{coil_cooling_water_to_air_heat_pump.name}, cannot find efficiency info using #{search_criteria}, cannot apply efficiency standard.")
+      successfully_set_all_properties = false
+      return successfully_set_all_properties
+    end
+
+    # Get the minimum efficiency standards
+    cop = nil
+
+    # If specified as EER (heat pump)
+    unless coil_props['minimum_full_load_efficiency'].nil?
+      min_eer = coil_props['minimum_full_load_efficiency']
+      cop = min_eer / 3.4121416
```

Comment:
@mdahlhausen , In the generated idf, I did a spot check, COP for Coil:Cooling:Watertoheatpump:Equationfit seems too low. It looks like this may be the cause. Please see the following I quoted from @jiangithub 

"For packaged (with fan and coils) heating and cooling HVAC equipment, the industry standards for efficiency rating includes the fan energy inputs in their efficiency rating, such as EER, SEER, heating COP. When we model such HVAC systems in EnergyPlus, we have separate fan and coil objects, in which their efficiency inputs are separate. So, the modeled COP input for a DX cooling coil is for the compressor only and does not include fan. Therefore, we should convert the efficiency metrics based on the industry standards to the modeled efficiency inputs. 

PNNL Method- From Thornton et al 2011. 

Thornton B.A., M.I. Rosenberg, E.E. Richman, W. Wang, Y. Xie, J. Zhang, and H. Cho, et al. 2011. Achieving the 30% Goal: Energy and Cost Savings Analysis of ASHRAE Standard 90.1-2010. PNNL-20405. Richland, WA: Pacific Northwest National Laboratory.

COP= (EER/3.413 + R) / (1-R)

EER = -0.0182 * SEER2 + 1.1088 * SEER"

Reply:
@leijerry888 I [changed the cooling coil](https://github.com/NREL/openstudio-standards/commit/338823e89ba62f6af2d6fa9d65394b173d9091f2) to use the [eer_to_cop](https://github.com/NREL/openstudio-standards/blob/master/lib/openstudio-standards/prototypes/common/objects/Prototype.utilities.rb#L289-L311) method that removes the fan from the cop accounting.  Should we do this for the heating coil as well?  I was hesitant to apply it there because that method [cop_heating_to_cop_heating_no_fan](https://github.com/NREL/openstudio-standards/blob/master/lib/openstudio-standards/prototypes/common/objects/Prototype.utilities.rb#L246-L253) assumes a COPH rated at 47F, rather than 68F for water-to-air coils.  Do you know what efficiency adjustment method we should use?

Reply:
Thanks @mdahlhausen . The heating coil COP looks good probably because 4.2 is the only option in there. @jiangithub , could you please confirm the answer to @mdahlhausen 's question? In the meantime, I will pull the new commit and test it.

Reply:
@mdahlhausen We didn't convert the rated heating COP in 90.1 to a COP input in the EnergyPlus Highrise Apartment. We took some manufacture specification data and developed a set of curves to make the COP and curve inputs to match the manufacture data and meet the COP in 90.1. 
I am not sure if [cop_heating_to_cop_heating_no_fan](https://github.com/NREL/openstudio-standards/blob/master/lib/openstudio-standards/prototypes/common/objects/Prototype.utilities.rb#L246-L253) method is an alternative way to do so. For the purpose of aligning the OpenStudio Standards prototypes with the EnergyPlus ones, let's use the same curves and use the COP in 90.1 and do not make any conversion.

Reply:
Results of a quick check: Comparing 2004 PNNL El_Paso v.s. 2004 OSSTD 3B: both have 80 coil:cooling:WaterToAirHeatPump:EquationFit objects. In PNNL model, 75/80 objects have COP of 3.86 and 5/80 have COP of 4.13; In OSSTD model, 59/80 objects have COP of 3.86 and 21/80 have COP of 4.13. I haven't looked into the cause.

Reply:
@mdahlhausen , I just discussed with @jiangithub , we think the COP differences mentioned in my previous comment are expected due to expected capacity differences between the PNNL and OSSTD models. As a sanity check at this moment, we think it is good as long as the calculated COP falls in one of the possible choices calculated based on the capacity.  I will check more model years and climate zones, and if all are good, I will commit the performance test results and regression models.

Reply:
@leijerry888 thanks for looking into this.  I expect the differences are from capacity.

For the curves - I didn't add coefficient methods or data yet, so these models are using the default OpenStudio/EnergyPlus coefficients.  Openstudio handles the curve coefficients for the WaterToAirHeatPump:EquationFit objects a bit differently by having coefficient fields in the object itself rather than as linked curves.  I can add the data/methods if you want that done in this pull request.

Reply:
@mdahlhausen The coefficients used in OSSTD models seem to be not default values, but from [here](https://github.com/NREL/openstudio-standards/blob/30818329a8a2db5e2da374cd93d4865c0e8fe7f1/lib/openstudio-standards/prototypes/common/objects/Prototype.CoilCoolingWaterToAirHeatPumpEquationFit.rb#L44-L59) and [here](https://github.com/NREL/openstudio-standards/blob/30818329a8a2db5e2da374cd93d4865c0e8fe7f1/lib/openstudio-standards/prototypes/common/objects/Prototype.CoilHeatingWaterToAirHeatPumpEquationFit.rb#L44-L53). These values match values in PNNL models. I am guessing it's because when creating these objects at [here](https://github.com/NREL/openstudio-standards/blob/4b4ae4b60ae9fc59bcd5b7a334c080ede24bfc15/lib/openstudio-standards/prototypes/common/objects/Prototype.hvac_systems.rb#L2836-L2841), the default type (i.e. nil) is passed in instead of "OS default"

Reply:
great!

Reply:
Checked all COP instances of OSSTD high rise apartment models. Cooling coils set COP to be one of [4.46473377194151, 3.86542098393842, 4.1317822230509, 4.19837253282902], all heating coils use COP of 4.2

These value match PNNL WSHP coil efficiency table values. Issue resolved. Regression models and performance test commit will come shortly